### PR TITLE
docs: add v4.3.21 and e4.3.16 change logs

### PR DESCRIPTION
**Pull request by zmstone**

### CODE REVIEW

**Review by hjianbo — DISMISSED** [review]

**Review by zhongwencool — DISMISSED** [review]

File: en_US/changes/changes-4.3.md [hunk 1]
```diff
@@ -1,21 +1,73 @@
...
+- TLS listener memory usage optimization [#9005](https://github.com/emqx/emqx/pull/9005).
+  New config `listener.ssl.$NAME.hibernate_after` to hibernate TLS connection process after idling.
+  Hibernation can reduce RAM usage significantly, but may cost more CPU.
+  This configuration is by default disabled.
+  Our preliminary test shows a 50% of RAM usage decline when configured to '5s'.
+
+- TLS listener default buffer size to 4KB [#9007](https://github.com/emqx/emqx/pull/9007)
+  Eliminate uncertainty that the buffer size is set by OS default.
+
+- Disable authorization for `api/v4/emqx_prometheus` endpoint. [8955](https://github.com/emqx/emqx/pull/8955)
```
Comment (JimMoen) [inline en_US/changes/changes-4.3.md]: #8955

File: en_US/changes/changes-4.3.md [hunk 2]
```diff
@@ -1,21 +1,73 @@
...
+
+### Bug fixes
+
+- Fix HTTP client library to handle SSL socket passive signal. [#9145](https://github.com/emqx/emqx/pull/9145)
+
+- Fix delayed publish inaccurate caused by os time change. [#8908](https://github.com/emqx/emqx/pull/8908)
+
+- Hide redis password in error logs [#9071](https://github.com/emqx/emqx/pull/9071)
+  In this change, it also included more changes in redis client:
+  - Improve redis connection error logging [eredis:19](https://github.com/emqx/eredis/pull/19).
```
Comment (JimMoen) [inline en_US/changes/changes-4.3.md]: use same style like line 45
eredis#19

File: zh_CN/changes/changes-4.3.md [hunk 3]
```diff
@@ -1,47 +1,82 @@
...
+
+- 修复延迟发布可能因为修改系统时间而导致的延迟等问题。 [#8908](https://github.com/emqx/emqx/pull/8908)
+
+### 修复
+
+- 修复 HTTP 客户端库启用 SSL 后 Socket 可能会进入 passive 状态。 [#9145](https://github.com/emqx/emqx/pull/9145)
+
+- 隐藏 redis 客户端错误日志中的密码参数 [#9071](https://github.com/emqx/emqx/pull/9071)
+  也包含了如下一些改进：
+  - 修复一些其他可能导致密码泄漏的隐患 [eredis:19](https://github.com/emqx/eredis/pull/19).
```
Comment (JimMoen) [inline zh_CN/changes/changes-4.3.md]: eredis#19

File: zh_CN/changes/changes-ee-4.3.md [hunk 4]
```diff
@@ -1,21 +1,86 @@
...
+- 所有的密码字段都使用敏感信息遮盖输入框。
+
+
+### 修复 (合并自 v4.3.21)
+
+- 修复 HTTP 客户端库启用 SSL 后 Socket 可能会进入 passive 状态。 [#9145](https://github.com/emqx/emqx/pull/9145)
+
+- 隐藏 redis 客户端错误日志中的密码参数 [#9071](https://github.com/emqx/emqx/pull/9071)
+  也包含了如下一些改进：
+  - 修复一些其他可能导致密码泄漏的隐患 [eredis:19](https://github.com/emqx/eredis/pull/19).
```
Comment (JimMoen) [inline zh_CN/changes/changes-ee-4.3.md]: eredis#19

File: zh_CN/changes/changes-4.3.md [hunk 5]
```diff
@@ -1,47 +1,82 @@
...
+  - 修复了 eredis_cluster 中连接池命名冲突的问题 [eredis_cluster#22](https://github.com/emqx/eredis_cluster/pull/22)
+    同时对这个库也进行了密码泄漏隐患对修复。
+
+- 修复共享订阅消息转发逻辑 [#9094](https://github.com/emqx/emqx/pull/9094).
+  - QoS2 飞行窗口消息丢弃的日志过度打印问题
+  - 对于通配符订阅的消息，会话中缓存的消息重发时，因为使用了发布主题（而非通配符主题）来进行重发，
+    导致无法匹配到 共享订阅组内的其他成员。
```
Comment (JimMoen) [inline zh_CN/changes/changes-4.3.md]: remove extra space

File: zh_CN/changes/changes-4.3.md [hunk 6]
```diff
@@ -1,47 +1,82 @@
...
+
+- 修复共享订阅 `sticky` 策略下，客户端取消订阅后仍然可以收到消息的问题。[#9119](https://github.com/emqx/emqx/pull/9119)
+  这之前的版本中，仅处理了客户端会话终结，而没有处理取消订阅。
+
+- 修复共享订阅 `sticky` 策略在某些情况下退化成 `random` 策略的问题。 [#9122](https://github.com/emqx/emqx/pull/9122)
+  集群环境下，在之前的版本中, 共享订阅组成员的选择在最开始时会随机选取，最终会选中在本节点连接的客户端，并开始粘性转发。
```
Comment (JimMoen) [inline zh_CN/changes/changes-4.3.md:47]: half-width comma.

File: en_US/changes/changes-ee-4.3.md [hunk 7]
```diff
@@ -1,21 +1,108 @@
...
+- Fix shared subscription 'sticky' strategy when there is no local subscriptions at all.
+  Prior to this change, it may take a few rounds to randomly pick group members until a local subscriber
+  is hit (and then start sticking to it).
+  After this fix, it will start sticking to whichever randomly picked member even when it is a
+  subscriber from another node in the cluster.
+  Fixed in [#9122](https://github.com/emqx/emqx/pull/9122)
+
+- Fix rule engine fallback actions metrics reset. [#9125](https://github.com/emqx/emqx/pull/9125)
+
+
```
Comment (id) [inline en_US/changes/changes-ee-4.3.md]: nit: extra line

File: en_US/changes/changes-4.3.md [hunk 8]
```diff
@@ -1,21 +1,73 @@
...
+- Fix shared subscription 'sticky' strategy when there is no local subscriptions at all.
+  Prior to this change, it may take a few rounds to randomly pick group members until a local subscriber
+  is hit (and then start sticking to it).
+  After this fix, it will start sticking to whichever randomly picked member even when it is a
+  subscriber from another node in the cluster.
+  Fixed in [#9122](https://github.com/emqx/emqx/pull/9122)
+
+- Fix rule engine fallback actions metrics reset. [#9125](https://github.com/emqx/emqx/pull/9125)
+
+
```
Comment (id) [inline en_US/changes/changes-4.3.md]: nit: extra line

File: en_US/changes/changes-4.3.md [hunk 9]
```diff
@@ -1,21 +1,73 @@
...
+  published when a client is denied connection. [#8894](https://github.com/emqx/emqx/pull/8894)
+
+- More rigorous checking of flapping to improve stability of the system. [#9045](https://github.com/emqx/emqx/pull/9045)
+
+- QoS1 and QoS2 messages in session's buffer are re-dispatched to other members in the group
+  when the session terminates [#9094](https://github.com/emqx/emqx/pull/9094).
+  Prior to this enhancement, one would have to set `broker.shared_dispatch_ack_enabled` to true
+  to prevent sessions from buffering messages, however this acknowledgement comes with a cost.
+
+
```
Comment (id) [inline en_US/changes/changes-4.3.md:31]: nit: extra line

**Review by id — DISMISSED** [review]

**Review by hjianbo — APPROVED** [review]

**Review by hjianbo — APPROVED** [review]

**Pull request merged**